### PR TITLE
Send Check First & Separate Out Errors

### LINKED ISSUE #38: Integration fails completly if initial setup fails
The integration does not recover when the initial setup fails.

Unfortunately, I was not able to generate more helpful information than this log output:
```
ERROR (MainThread) [custom_components.healthchecksio] Could not update data -
```
Which seems to come from here:
https://github.com/custom-components/healthchecksio/blob/main/custom_components/healthchecksio/__init__.py#L118

On my Raspberry Pi 3 with HASSOS it fails to initialize the Integration about every second time and I have not yet been able to identify the problem. It seems that a simple retry should help but I'm not sure how I could force it to do so. I've even added a job that updates the sensory every 2 minutes.

### PR DESCRIPTION
**Pull request by Snuffy2**
* Send the Check first before getting the data from HealthCheck. That way, the Check should be reflected in the Data downloaded.
* Separate out the error checks to be more discrete. Should give more specific errors and allow the integration to fail more gracefully so it will continue to update every 5 min even if there is an error

Fixes #38

### CODE REVIEW

File: custom_components/healthchecksio/config_flow.py [hunk 1]
```diff
@@ -123,23 +125,58 @@ async def _test_credentials(
...
+            if data.ok:
+                Logger("custom_components.healthchecksio").debug(
+                    f"Get Data HTTP Status Code: {data.status}"
+                )
+                return True
+            else:
+                Logger("custom_components.healthchecksio").error(
+                    f"Error: Get Data HTTP Status Code: {data.status}"
+                )
+                return False
```
Comment (ludeeus) [inline custom_components/healthchecksio/config_flow.py:182]:
```suggestion
            if not data.ok:
                Logger("custom_components.healthchecksio").error(
                    f"Error: Get Data HTTP Status Code: {data.status}"
                )
                return False
            Logger("custom_components.healthchecksio").debug(
                f"Get Data HTTP Status Code: {data.status}"
            )
            return True
```

File: custom_components/healthchecksio/__init__.py [hunk 2]
```diff
@@ -95,28 +92,56 @@ def __init__(self, hass, api_key, check, self_hosted, site_root, ping_endpoint):
...
+            else:
+                Logger("custom_components.healthchecksio").error(
+                    f"Error: Send Check HTTP Status Code: {check_response.status}"
+                )
+        try:
+            async with session.get(
+                f"{self.site_root}/api/v1/checks/",
+                headers=headers,
+                timeout=timeout10,
+            ) as data:
```
Comment (ludeeus) [inline custom_components/healthchecksio/__init__.py:126]:
```suggestion
            data = await session.get(
                f"{self.site_root}/api/v1/checks/",
                headers=headers,
                timeout=timeout10,
            )
```

File: custom_components/healthchecksio/config_flow.py [hunk 3]
```diff
@@ -123,23 +125,58 @@ async def _test_credentials(
...
+                Logger("custom_components.healthchecksio").error(
+                    f"Error: Send Check HTTP Status Code: {check_response.status}"
+                )
+                return False
+        try:
+            async with session.get(
+                f"{site_root}/api/v1/checks/",
+                headers=headers,
+                timeout=timeout10,
+            ) as data:
```
Comment (ludeeus) [inline custom_components/healthchecksio/config_flow.py:160]:
```suggestion
            data = await session.get(
                f"{site_root}/api/v1/checks/",
                headers=headers,
                timeout=timeout10,
            )
```

File: custom_components/healthchecksio/config_flow.py [hunk 4]
```diff
@@ -123,23 +125,58 @@ async def _test_credentials(
...
                 self.hass.data[DOMAIN_DATA] = {"data": await data.json()}
```
Comment (ludeeus) [inline custom_components/healthchecksio/config_flow.py:161]:
```suggestion
            self.hass.data[DOMAIN_DATA] = {"data": await data.json()}
```

File: custom_components/healthchecksio/__init__.py [hunk 5]
```diff
@@ -95,28 +92,56 @@ def __init__(self, hass, api_key, check, self_hosted, site_root, ping_endpoint):
...
                 self.hass.data[DOMAIN_DATA]["data"] = await data.json()
```
Comment (ludeeus) [inline custom_components/healthchecksio/__init__.py:127]:
```suggestion
            self.hass.data[DOMAIN_DATA]["data"] = await data.json()
```

**Review by ludeeus — CHANGES_REQUESTED** [review]

**Review by Snuffy2 — COMMENTED** [review]
All requested changes implemented.

**Comment by Snuffy2** [comment]
@ludeeus, any chance of getting this and #46 reviewed/merged?

**Review by ludeeus — APPROVED** [review]

**Pull request merged**